source/extensions/access_loggers/file/config.cc

```diff
@@ -29,30 +30,33 @@ FileAccessLogFactory::createAccessLogInstance(const Protobuf::Message& config,
       config, context.messageValidationVisitor());
   AccessLog::FormatterPtr formatter;
 
-  if (fal_config.access_log_format_case() == envoy::extensions::access_loggers::file::v3::
-                                                 FileAccessLog::AccessLogFormatCase::kFormat ||
-      fal_config.access_log_format_case() ==
-          envoy::extensions::access_loggers::file::v3::FileAccessLog::AccessLogFormatCase::
-              ACCESS_LOG_FORMAT_NOT_SET) {
-    if (fal_config.format().empty()) {
-      formatter = AccessLog::AccessLogFormatUtils::defaultAccessLogFormatter();
-    } else {
-      formatter = std::make_unique<AccessLog::FormatterImpl>(fal_config.format());
+  if (fal_config.has_log_format()) {
+    formatter = SubstitutionFormatStringUtils::fromProtoConfig(fal_config.log_format());
+  }
+  // backward compatible code for deprecated access_log_format, to be removed.
+  if (fal_config.access_log_format_case() !=
+      envoy::extensions::access_loggers::file::v3::FileAccessLog::AccessLogFormatCase::
+          ACCESS_LOG_FORMAT_NOT_SET) {
+    envoy::config::core::v3::SubstitutionFormatString sff_config;
+    switch (fal_config.access_log_format_case()) {
+    case envoy::extensions::access_loggers::file::v3::FileAccessLog::AccessLogFormatCase::kFormat:
+      sff_config.set_text_format(fal_config.format());
+      break;
+    case envoy::extensions::access_loggers::file::v3::FileAccessLog::AccessLogFormatCase::
+        kJsonFormat:
+      *sff_config.mutable_json_format() = fal_config.json_format();
+      break;
+    case envoy::extensions::access_loggers::file::v3::FileAccessLog::AccessLogFormatCase::
+        kTypedJsonFormat:
+      *sff_config.mutable_typed_json_format() = fal_config.typed_json_format();
+      break;
+    default:
+      break;
```

Comment:
Per other comment I think NOT_REACHED if there is a validator?

Reply:
Done